LibZipSharp/Xamarin.Tools.Zip/ZipArchive.cs

```diff
@@ -789,7 +789,9 @@ internal static unsafe Int64 stream_callback (IntPtr state, IntPtr data, UInt64
 				case SourceCommand.SeekWrite:
 				case SourceCommand.Seek:
 					Native.zip_error_t error;
-					UInt64 offset = Native.zip_source_seek_compute_offset ((UInt64)stream.Position, (UInt64)stream.Length, data, len, out error);
+					Int64 offset = Native.zip_source_seek_compute_offset ((UInt64)stream.Position, (UInt64)stream.Length, data, len, out error);
+					if (offset == -1)
```

Comment:
I wonder if we should make it `offset < 0`, despite the documentation stating that the function returns `-1` on error. The computed offset appears to be absolute, so any negative value should be treated as an error IMO.